Server fails to start if all enchantments are removed due to item obfuscation trying to find one

### Stack trace

```
Caused by: java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.orElseThrow(Optional.java:377) ~[?:?]
        at io.papermc.paper.util.ItemComponentSanitizer.dummyEnchantments(ItemComponentSanitizer.java:59) ~[paper-1.21.4.jar:1.21.4-151-9b9de82]
        at io.papermc.paper.util.ItemComponentSanitizer.lambda$static$0(ItemComponentSanitizer.java:30) ~[paper-1.21.4.jar:1.21.4-151-9b9de82]
        at net.minecraft.Util.make(Util.java:596) ~[paper-1.21.4.jar:1.21.4-151-9b9de82]
        at io.papermc.paper.util.ItemComponentSanitizer.<clinit>(ItemComponentSanitizer.java:28) ~[paper-1.21.4.jar:1.21.4-151-9b9de82]
```
https://pastes.dev/Zt43fOLbEs


### Plugin and Datapack List

Disabled the vanilla datapack & removed all enchantments from mine

### Actions to reproduce (if known)

1) Disable the vanilla datapack
2) Add a new datapack that contains the things vanilla requires (tags, some worldgen etc), but crucially no enchantments. I've attached a datapack that does this here: [Erethon.zip](https://github.com/user-attachments/files/18812228/Erethon.zip)
3) Try starting the server. It will crash once it attempts to load the Paper global config

### Paper version

1.21.4-151

### Other

The dummy enchantment methods tries to look up a random one, and obviously fails if there aren't any.
```
    private static ItemEnchantments dummyEnchantments() {
        final ItemEnchantments.Mutable obj = new ItemEnchantments.Mutable(ItemEnchantments.EMPTY);
        obj.set(MinecraftServer.getServer().registryAccess().lookupOrThrow(Registries.ENCHANTMENT).getRandom(RandomSource.create()).orElseThrow(), 1);
        return obj.toImmutable();
    }
```

--- a/paper-server/src/main/java/io/papermc/paper/util/ItemComponentSanitizer.java
+++ b/paper-server/src/main/java/io/papermc/paper/util/ItemComponentSanitizer.java
@@ -27,9 +27,12 @@ public final class ItemComponentSanitizer {
      */
     static final Map<DataComponentType<?>, UnaryOperator<?>> SANITIZATION_OVERRIDES = Util.make(ImmutableMap.<DataComponentType<?>, UnaryOperator<?>>builder(), (map) -> {
             put(map, DataComponents.LODESTONE_TRACKER, empty(new LodestoneTracker(Optional.empty(), false))); // We need it to be present to keep the glint
-            put(map, DataComponents.ENCHANTMENTS, empty(dummyEnchantments())); // We need to keep it present to keep the glint
-            put(map, DataComponents.STORED_ENCHANTMENTS, empty(dummyEnchantments())); // We need to keep it present to keep the glint
             put(map, DataComponents.POTION_CONTENTS, ItemComponentSanitizer::sanitizePotionContents); // Custom situational serialization
+
+            if (MinecraftServer.getServer().registryAccess().lookupOrThrow(Registries.ENCHANTMENT).size() > 0) {
+                put(map, DataComponents.ENCHANTMENTS, empty(dummyEnchantments())); // We need to keep it present to keep the glint
+                put(map, DataComponents.STORED_ENCHANTMENTS, empty(dummyEnchantments())); // We need to keep it present to keep the glint
+            }
         }
     ).build();
 

